src/main/java/com/atlassian/db/replica/api/AuroraMultiReplicaConsistency.java

```diff
@@ -39,12 +40,16 @@ public void write(Connection main) {
     public boolean isConsistent(Supplier<Connection> replicaSupplier) {
         Collection<Database> replicas = cluster.getReplicas(replicaSupplier);
 
-        if (replicas.isEmpty()) {
-            return false;
-        }
-
         return replicas.stream()
-            .allMatch(replica -> replicaConsistency.isConsistent(() -> replica.getConnectionSupplier().get()));
+            .map(replica -> {
+                try (Connection connection = replica.getConnectionSupplier().get()) {
+                    return replicaConsistency.isConsistent(() -> connection);
+                } catch (SQLException exception) {
+                    return false;
```

Comment:
I'm open to suggestions as to what should happen when this exception is caught

Reply:
We can not catch the exception, and allow the API caller to decide. 🤔

Reply:
IMO Re-throw as runtime exception

Reply:
Why RE instead of SQLE?

Reply:
Checked exceptions inside lambda are not allowed AFAIK

Reply:
we cannot rethrow as SQLE because be cannot change the signature of the `isConsistent` method

Reply:
unless we added it in the `ReplicaConsistency` interface as well

Reply:
In the future we could return some useful object instead of a dummy boolean. It could contain all replicas with boolean/list of errors.

Reply:
@dwatl Remember `ReplicaConsistency` is SPI. DualConnection uses it to determine if the replica is consistent or not. Still, we'll need to decide what to do with the exception inside of the library. Currently, it's re-thrown.
```
try {
            isConsistent = consistency.isConsistent(() -> readConnection.get(decisionBuilder));
        } catch (Exception e) {
            closeConnection(readConnection, decisionBuilder);
            throw e;
        }
```

> It could contain all replicas with boolean/list of errors.

Multi-replica lives only in one `ReplicaConsistency` implementation. Returning data specific only for a single implementation from API would make a poor abstraction.